### ISSUE TITLE
different generated result with huggingface generate interface

### ISSUE THREAD

**Issue by MasKong** [abetlen/llama-cpp-python#441]
It seems that the result of llama-cpp is different from huggingface. Is it expected?

#### huggingface generate code:
```
prompt = "Tell me about Shanghai"
max_new_tokens = 32
generation_config = GenerationConfig(
    top_k=1, top_p=1, temp=0,
)
start_time = time.time()
with torch.no_grad():
    batch_generation_output = model.generate(
        input_ids=input_ids,
        generation_config=generation_config,
        return_dict_in_generate=True,
        output_scores=True,
        max_new_tokens=max_new_tokens,
        attention_mask=attn_masks,
    )
```

#### llama-cpp-python code
```
output = llm.create_completion(
    " Tell me about Shanghai",
    stop=["</s>", ],
    #stop=["</s>", "\n"],
    max_tokens=32,
    echo=True,
    temperature=0,
    top_k=1, top_p=1
)
```

### result
huggingface output: `I'm going to Shanghai in a few weeks and I'm looking for some advice. I'm going to be there for a week`

llama-cpp-python output:  `I'm not sure what you mean by that.\n I'll tell you about it later.\n I'd like to know more about your family.\n I'd like to know more about the company.`

# Environment and Context
llama-cpp-python 0.1.67
transformers 4.30.2
torch 2.0
cuda 11.2
Ubuntu 18.04
llama model: https://huggingface.co/huggyllama/llama-7b/
```
# convert the 7B model to ggml FP16 format with convert.py from llama-cpp
python3 convert.py models/7B/

**Comment by poohzaza166** [comment]
have you try using the same seed?

**Comment by gjmulder** [comment]
[#386](https://github.com/abetlen/llama-cpp-python/issues/386)

**Comment by MasKong** [comment]
> [#386](https://github.com/abetlen/llama-cpp-python/issues/386)

Still not working. I have set seed to 0. And huggingface default generation doesn't use seed. 
By the way, top_k = 1 means no sample. Does seed matter under this circumstance?

**Comment by luanshaotong** [comment]
Same problem between llama-cpp-python and llama-cpp.

**Comment by abetlen** [comment]
Very late reply but in case anyone finds this, yes it's very likely that you'll encounter different generations for the same model, this can be caused by both quantisation introducting differences in the model weights as well as simple implementation differences that cause different floating point operations to be occur in different orders leading to precision errors.
